code/game/objects/items/devices/tvcamera.dm

```diff
@@ -0,0 +1,147 @@
+	camera.c_tag = channel
+	camera.status = FALSE
+	radio = new(src)
+	radio.get_frequency(FALSE)
+	radio.set_frequency(ENT_FREQ)
+	GLOB.listening_objects += src
+	. = ..()
+
+/obj/item/device/tvcamera/examine(mob/user)
+	. = ..()
+	to_chat(user, "Video feed is currently: [camera.status ? "Online" : "Offline"]")
+	to_chat(user, "Audio feed is currently: [radio.get_broadcasting() ? "Online" : "Offline"]")
+
+/obj/item/device/tvcamera/attack_self(mob/user)
+	add_fingerprint(user)
+	user.set_machine(src)
+	var/dat = list()
+	dat += "Channel name is: <a href='?src=\ref[src];channel=1'>[channel ? channel : "unidentified broadcast"]</a><br>"
+	dat += "Video streaming is: <a href='?src=\ref[src];video=1'>[camera.status ? "Online" : "Offline"]</a><br>"
+	dat += "Microphone is: <a href='?src=\ref[src];sound=1'>[radio.get_broadcasting() ? "Online" : "Offline"]</a><br>"
+	dat += "Sound is being broadcasted on frequency: [format_frequency(radio.get_frequency())] ([get_frequency_name(radio.get_frequency())])<br>"
+	var/datum/browser/popup = new(user, "Press Camera Drone", "EyeBuddy", 300, 390, src)
+	popup.set_content(jointext(dat,null))
+	popup.open()
+
+/obj/item/device/tvcamera/Topic(bred, href_list, state = physical_state)
+	if(..())
+		return 1
+	if(href_list["channel"])
+		var/nc = sanitize(input(usr, "Channel name", "Select new channel name", channel) as text|null)
+		if(nc)
+			channel = nc
+			camera.c_tag = channel
+			to_chat(usr, "<span class='notice'>New channel name: '[channel]' has been set.</span>")
+	if(href_list["video"])
+		camera.set_status(!camera.status)
+		if(camera.status)
+			to_chat(usr,"<span class='notice'>Video streaming: Activated. Broadcasting on channel: '[channel]'</span>")
+		else
+			to_chat(usr,"<span class='notice'>Video streaming: Deactivated.</span>")
+		update_icon()
+	if(href_list["sound"])
+		radio.set_broadcasting(!radio.get_broadcasting())
+		if(radio.get_broadcasting())
+			to_chat(usr,"<span class='notice'>Audio streaming: Activated. Broadcasting on frequency: [format_frequency(radio.get_frequency())].</span>")
+		else
+			to_chat(usr,"<span class='notice'>Audio streaming: Deactivated.</span>")
+	if(!href_list["close"])
+		attack_self(usr)
+
+/obj/item/device/tvcamera/update_icon()
+	..()
+	if(camera.status)
+		icon_state = "camcorder_on"
+		item_state = "camcorder_on"
+	else
+		icon_state = "camcorder"
+		item_state = "camcorder"
+	var/mob/living/carbon/human/H = loc
+	if(istype(H))
+		H.update_inv_r_hand(0)
+		H.update_inv_l_hand()
+
+/* Assembly by a roboticist */
+/obj/item/robot_parts/head/attackby(var/obj/item/device/assembly/S, mob/user as mob)
+	if(!istype(S, /obj/item/device/assembly/infra))
+		..()
+		return
+	var/obj/item/TVAssembly/A = new(user)
+	qdel(S)
+	user.put_in_hands(A)
+	to_chat(user, "<span class='notice'>You add the infrared sensor to the robot head.</span>")
+	qdel(src)
+
+/* Using camcorder icon as I can't sprite.
+Using robohead because of restricting to roboticist */
+/obj/item/TVAssembly
```

Comment:
Typepaths with capitals like this aren't ideal. Name it `tv_assembly` or `camcorder_assembly` or something.